src/PMGDIterators.h

```diff
@@ -132,4 +134,152 @@ namespace VDMS {
 
         PMGD::Node *ref() { return &**_neighb_i; }
     };
+
+    class PMGDQueryHandler::NodeEdgeIteratorImpl : public PMGD::EdgeIteratorImplIntf
```

Comment:
I need to understand the purpose of this and how it is used in order to review it.